### PR TITLE
Refactor /results to use joins with db.

### PR DESCRIPTION
**Pull request by Sorsby**
Change results to remove multiple db calls, replaced with a single call to the database using joins.
Updated the tests to reflect the implementation.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (MikeSilverstone) [inline .vscode/settings.json]: Can we put this file in the .gitignore and remove it.

File: api/db/models.py [hunk 2]
```diff
@@ -72,13 +86,17 @@ def get_round(self, id):
 class RaceResult(db.Model):
     id = db.Column(db.Integer(), primary_key=True, autoincrement=True)
     position = db.Column(db.Integer(), nullable=False)
-    time_to_finish = db.Column(db.DateTime(), nullable=False)
-    did_not_finish = db.Column(db.Boolean(), nullable=False)
-    id_race_participants = db.Column(db.Integer(), db.ForeignKey("race_participants.id"))
+    time_to_finish = db.Column(db.Integer(), nullable=True)
+    did_not_finish = db.Column(db.String(3))
```
Comment (MikeSilverstone) [inline api/db/models.py]: Should DNF be a boolean? If not, why is it a three letter string?

File: api/endpoints/results/results.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class ResultRace:
+
+    def __init__(self, id):
+        self.id = id
+        self.snails = []
+        self.json = []
+
+    def __eq__(self, other):
```
Comment (MikeSilverstone) [inline api/endpoints/results/results.py]: These can go

File: api/endpoints/results/results.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+class ResultSnail:
+
+    def __init__(self, id, pos, time, dnf):
+        self.id = id
+        self.pos = pos
+        self.time = time
+        self.dnf = dnf
+
+    def __eq__(self, other):
```
Comment (MikeSilverstone) [inline api/endpoints/results/results.py]: Likewise

File: .vscode/settings.json (hunk 1 above)
Comment (Sorsby) [reply .vscode/settings.json]: I have merged develop into this branch to fix conflicts, this is resolved in that merge.

File: api/db/models.py (hunk 2 above)
Comment (Sorsby) [reply api/db/models.py]: Fixed in merge.

**Pull request merged**